ext/odbc/odbc.stub.php

```diff
@@ -4,193 +4,193 @@
+/** @param resource $stmt */
+function odbc_longreadlen($stmt, int $length): bool {}
 
 /**
- * @param resource $connection_id
+ * @param resource $odbc
  * @return resource|false
  */
-function odbc_prepare($connection_id, string $query) {}
+function odbc_prepare($odbc, string $query) {}
 
-/** @param resource $result_id */
-function odbc_execute($result_id, array $parameters_array = []): bool {}
+/** @param resource $stmt */
+function odbc_execute($stmt, array $params = []): bool {}
 
-/** @param resource $result_id */
-function odbc_cursor($result_id): string|false {}
+/** @param resource $stmt */
+function odbc_cursor($stmt): string|false {}
 
 #ifdef HAVE_SQLDATASOURCES
-/** @param resource $connection_id */
-function odbc_data_source($connection_id, int $fetch_type): array|false {}
+/** @param resource $odbc */
+function odbc_data_source($odbc, int $fetch_type): array|false {}
 #endif
 
 /**
- * @param resource $connection_id
+ * @param resource $odbc
  * @return resource|false
  */
-function odbc_exec($connection_id, string $query) {}
+function odbc_exec($odbc, string $query) {}
 
 /**
- * @param resource $connection_id
+ * @param resource $odbc
  * @return resource|false
  * @alias odbc_exec
  */
-function odbc_do($connection_id, string $query) {}
+function odbc_do($odbc, string $query) {}
 
 #ifdef PHP_ODBC_HAVE_FETCH_HASH
-/** @param resource $result */
-function odbc_fetch_object($result, int $rownumber = -1): stdClass|false {}
+/** @param resource $stmt */
+function odbc_fetch_object($stmt, int $row_number = -1): stdClass|false {}
 
-/** @param resource $result */
-function odbc_fetch_array($result, int $rownumber = -1): array|false {}
+/** @param resource $stmt */
+function odbc_fetch_array($stmt, int $row_number = -1): array|false {}
 #endif
 
 /**
- * @param resource $result_id
- * @param array $result_array
+ * @param resource $stmt
+ * @param array $row
  */
-function odbc_fetch_into($result_id, &$result_array, int $rownumber = 0): int|false {}
+function odbc_fetch_into($stmt, &$row, int $row_number = 0): int|false {}
 
-/** @param resource $result_id */
-function odbc_fetch_row($result_id, ?int $row_number = null): bool {}
+/** @param resource $stmt */
+function odbc_fetch_row($stmt, ?int $row_number = null): bool {}
 
-/** @param resource $result_id */
-function odbc_result($result_id, string|int $field): string|bool|null {}
+/** @param resource $stmt */
+function odbc_result($stmt, string|int $column): string|bool|null {}
 
-/** @param resource $result_id */
-function odbc_result_all($result_id, string $format = ""): int|false {}
+/** @param resource $stmt */
+function odbc_result_all($stmt, string $format = ""): int|false {}
 
-/** @param resource $result_id */
-function odbc_free_result($result_id): bool {}
+/** @param resource $stmt */
+function odbc_free_result($stmt): bool {}
 
 /** @return resource|false */
 function odbc_connect(string $dsn, string $user, string $password, int $cursor_option = SQL_CUR_USE_DRIVER) {}
```

Comment:
```suggestion
function odbc_connect(string $dsn, string $user, string $password, int $cursor_type = SQL_CUR_USE_DRIVER) {}
```
More in line with $fetch_type above?

Reply:
`$cursor_type` appears to be misleading; this is not about `SQL_CURSOR_FORWARD_ONLY`, `SQL_CURSOR_KEYSET_DRIVEN`, `SQL_CURSOR_DYNAMIC` or `SQL_CURSOR_STATIC`, but rather whether the ODBC cursor library (a compat layer for minimal drivers, which is deprecated at least on Windows) should be used. I don't have a better idea, so would be okay with `$cursor_type`).